### ISSUE TITLE
Add the ability to download the content (print documents)

### ISSUE THREAD

**Issue by djplaner** [djplaner/Content-Interface-Tweak#19]
Explore if Javascript librararies might provide a way to produce offline versions of Content Interface.

And/or Python libraries that  might be combined with screen scraping.

### Rationale

The Word documents can contain styling and embedded documents that don't print well.  They are applied on the web.  If we're able to use those to generate PDF/DOC the styling will be applied

### Current status

Weasyprint with Python script can produce a PDF that's close, but

- Needs Bb content interaface to be set to expandAll
- Doesn't handle some of the CSS alignment well (e.g. exerises, pictureRight)
- Doesn't handle YouTube videos

**Comment by djplaner** [comment]
[Comparison](https://gearheart.io/blog/how-generate-pdf-files-python-xhtml2pdf-weasyprint-or-unoconv/) of XHTML2PDF, WeasyPrint and UnoConv

## Possible options

 ### [WeasyPrint](https://weasyprint.org/)

Python - [source code](https://github.com/Kozea/WeasyPrint)
[Documentation](https://weasyprint.readthedocs.io/en/stable/tutorial.html) 

Seems to be able to be provided a HTML string.

### [XHTML2PDF](https://pypi.org/project/xhtml2pdf/)

Perhaps not as complete.

### UnoConv

Is based on using OpenOffice.  COuld be interetsing, but heavyweight.

**Comment by djplaner** [comment]
### WeasyPrint

1. Install locally - **Success**
> pip install Weasyprint 
1. Try to print [edu8702 week1](https://djon.es/edu8702/week1.html) 
> weasyprint _URL_ _PDFFile_
> **Failure on Windows** with default install
1. Try it on Linux **success**
1. Try it with some content from ContentInterface
> **success** somewhat.  Still need to run the Javascript on it.
1. Try it with Javascript enabled
> **FAILURE** The javascript is not handled by Weasyprint - at least default


### Handling javascript

[Suggestion](https://github.com/Kozea/WeasyPrint/issues/454) is to use a Javascript pre-processor first.

Apparently [PhantomJS ](https://phantomjs.org/) is an option. But it isn't being maintained.  The alternative is [headless chrome](https://developers.google.com/web/updates/2017/04/headless-chrome) or similar. i.e. something that works on the completed DOM after Javascript has run.  e.g. [Python + Selenium](https://realpython.com/modern-web-automation-with-python-and-selenium/)

Basically working.  However, by default accordions can create some issues.  Remove them and all ok.

What about expandAll?  Could get selenium to click the expandAll button


### How to show iframes/YouTube videos

These don't seem to working too well.
